### PR TITLE
Also remove 389-ds LDAP server

### PR DESCRIPTION
**Pull request by jcpunk**
#### Description:

RHEL8 dropped OpenLDAP, so add a similar check to `openldap-servers` for `389-ds` server.

#### Rationale:

The identical reasons for removing `openldap-servers` applies to `389-ds-base` as it is also an LDAP server.

### CODE REVIEW

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+documentation_complete: true
+  
+prodtype: rhel6,rhel7,rhel8
+
+title: 'Uninstall 389-ds-base Package'
+
+description: |-
+    The <tt>389-ds-base</tt> package should be removed if not in use.
+    Is this system the 389-ds server? If not, remove the package.
```
Comment (jan-cerny) [inline linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: This line seems to be superfluous.

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    disa: "366"
+    nist: CM-7(a),CM-7(b),CM-6(a)
+    nist-csf: PR.IP-1,PR.PT-3
+    isa-62443-2013: 'SR 1.1,SR 1.10,SR 1.11,SR 1.12,SR 1.13,SR 1.2,SR 1.3,SR 1.4,SR 1.5,SR 1.6,SR 1.7,SR 1.8,SR 1.9,SR 2.1,SR 2.2,SR 2.3,SR 2.4,SR 2.5,SR 2.6,SR 2.7,SR 7.6'
+    isa-62443-2009: 4.3.3.5.1,4.3.3.5.2,4.3.3.5.3,4.3.3.5.4,4.3.3.5.5,4.3.3.5.6,4.3.3.5.7,4.3.3.5.8,4.3.3.6.1,4.3.3.6.2,4.3.3.6.3,4.3.3.6.4,4.3.3.6.5,4.3.3.6.6,4.3.3.6.7,4.3.3.6.8,4.3.3.6.9,4.3.3.7.1,4.3.3.7.2,4.3.3.7.3,4.3.3.7.4,4.3.4.3.2,4.3.4.3.3
+    cobit5: BAI10.01,BAI10.02,BAI10.03,BAI10.05,DSS05.02,DSS05.05,DSS06.06
+    iso27001-2013: A.12.1.2,A.12.5.1,A.12.6.2,A.14.2.2,A.14.2.3,A.14.2.4,A.9.1.2
+    cis-csc: 11,14,3,9
+
+ocil_clause: 'it does not'
```
Comment (jan-cerny) [inline linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]:
```suggestion
ocil_clause: 'the package is installed'
```

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    stigid@rhel6: "000256"
+    srg@rhel6: SRG-OS-999999
+    disa: "366"
+    nist: CM-7(a),CM-7(b),CM-6(a)
+    nist-csf: PR.IP-1,PR.PT-3
+    isa-62443-2013: 'SR 1.1,SR 1.10,SR 1.11,SR 1.12,SR 1.13,SR 1.2,SR 1.3,SR 1.4,SR 1.5,SR 1.6,SR 1.7,SR 1.8,SR 1.9,SR 2.1,SR 2.2,SR 2.3,SR 2.4,SR 2.5,SR 2.6,SR 2.7,SR 7.6'
+    isa-62443-2009: 4.3.3.5.1,4.3.3.5.2,4.3.3.5.3,4.3.3.5.4,4.3.3.5.5,4.3.3.5.6,4.3.3.5.7,4.3.3.5.8,4.3.3.6.1,4.3.3.6.2,4.3.3.6.3,4.3.3.6.4,4.3.3.6.5,4.3.3.6.6,4.3.3.6.7,4.3.3.6.8,4.3.3.6.9,4.3.3.7.1,4.3.3.7.2,4.3.3.7.3,4.3.3.7.4,4.3.4.3.2,4.3.4.3.3
+    cobit5: BAI10.01,BAI10.02,BAI10.03,BAI10.05,DSS05.02,DSS05.05,DSS06.06
+    iso27001-2013: A.12.1.2,A.12.5.1,A.12.6.2,A.14.2.2,A.14.2.3,A.14.2.4,A.9.1.2
+    cis-csc: 11,14,3,9
```
[the remark this answers is not in the document]
Comment (jcpunk) [reply linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: Those were lifted from the OpenLDAP definition.  I'm certain that, if the OpenLDAP section is right these are also.  Any justification that covers OpenLDAP should apply identically to 389-ds.

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    Unnecessary packages should not be installed to decrease the attack
+    surface of the system.  While this software is clearly essential on an LDAP
+    server, it is not necessary on typical desktop or workstation systems.
+
+severity: low
+
+identifiers:
+    cce@rhel6: 26858-1
+    cce@rhel7: 80293-4
+    cce@rhel8: 82415-1
```
[the remark this answers is not in the document]
Comment (jcpunk) [reply linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: I'll pull from there shortly.

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml (hunk 2 above)
Comment (jcpunk) [reply linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: I'll switch this and rebase shortly.

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml (hunk 1 above)
Comment (jcpunk) [reply linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: It was lifted from the OpenLDAP rule.  I can drop it if you'd like.

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    Unnecessary packages should not be installed to decrease the attack
+    surface of the system.  While this software is clearly essential on an LDAP
+    server, it is not necessary on typical desktop or workstation systems.
+
+severity: low
+
+identifiers:
+    cce@rhel6: 82722-0
+    cce@rhel7: 82723-8
+    cce@rhel8: 82724-6
```
Comment (shawndwells) [inline linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: CCEs! Nice! Please also add one for ``ocp4``, which will also cover CoreOS.

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (shawndwells) [reply linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: Seems like a fair assumption.

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml (hunk 1 above)
Comment (shawndwells) [reply linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: The same sentiment is expressed later through lines 11-14. Suggest dropping the ``Is the system the ....`` part.

Bonus points if you feel like cleaning up the original OpenLDAP language too! Won't hold up a merge if not.

File: linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml (hunk 5 above)
Comment (jcpunk) [reply linux_os/guide/services/ldap/389_ds/package_389-ds-base_removed/rule.yml]: Can do.

**Comment by jcpunk** [comment]
I went with the bonus points and cleaned up OpenLDAP while I was there....

**Comment by jan-cerny** [comment]
@jcpunk The patch looks good to me, but unfortunately we have got a conflict in shared/references/cce-redhat-avail.txt. Please rebase your branch.

**Comment by jcpunk** [comment]
Rebased.

**Pull request merged**